src/ts_catalog/continuous_aggs_watermark.c

```diff
@@ -88,6 +89,21 @@ cagg_watermark_get(Hypertable *mat_ht)
 	ScanIterator iterator =
 		ts_scan_iterator_create(CONTINUOUS_AGGS_WATERMARK, AccessShareLock, CurrentMemoryContext);
 
+	/*
+	 * The watermark of a CAGG has to be fetched by using the active snapshot.
+	 *
+	 * The watermark of a CAGG can be read by multiple parallel workers (i.e., in ChunkAppend). All
+	 * of the workers have to see the same watermark in order to act consistently (e.g., when
+	 * excluding chunks). If no snapshot is specified, there is a chance that one worker may see a
+	 * different watermark than another, if the CAGG was updated in between their reads (e.g., read
+	 * of worker a, CAGG update, read of worker B).
+	 *
+	 * PostgreSQL synchronizes the snapshots between the parallel worker. Using the same snapshot
+	 * ensures that all parallel workers have the same view of the stored watermark.
+	 */
+	iterator.ctx.snapshot = GetActiveSnapshot();
+	Assert(iterator.ctx.snapshot != NULL);
```

Comment:
```suggestion
	if (ActiveSnapshotSet())
		iterator.ctx.snapshot = GetActiveSnapshot();
```
What about set the active snapshot just when we have an active snapshot?

Reply:
I had also thought about a similar code, but then discarded it. Shouldn't we require that this method can only be called with an active snapshot?